engine/src/main/java/org/hibernate/search/indexes/impl/IndexShardingStrategyIndexManagerSelector.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.indexes.impl;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.apache.lucene.document.Document;
+import org.hibernate.search.filter.FullTextFilterImplementor;
+import org.hibernate.search.indexes.spi.IndexManager;
+import org.hibernate.search.indexes.spi.IndexManagerSelector;
+import org.hibernate.search.spi.IndexedTypeIdentifier;
+import org.hibernate.search.store.IndexShardingStrategy;
+
+
+/**
+ * @author Yoann Rodiere
+ */
+public class IndexShardingStrategyIndexManagerSelector implements IndexManagerSelector {
+
+	private final IndexShardingStrategy shardingStrategy;
+	private final IndexManager[] allIndexManagers;
+
+	public IndexShardingStrategyIndexManagerSelector(IndexShardingStrategy shardingStrategy, IndexManager[] allIndexManagers) {
+		super();
+		this.shardingStrategy = shardingStrategy;
+		this.allIndexManagers = allIndexManagers;
+	}
+
+	@Override
+	public Set<IndexManager> all() {
```

Comment:
Sure. That's the main difference between the dynamic sharding strategies and the "legacy" (this one)